### ISSUE TITLE
Question about guidance of optimizing

### ISSUE THREAD

**Issue by mawangdan** [scenediffuser/Scene-Diffuser#26]
I noticed that in pose_in_scene.py, the parameter x of the optimize function: torch.Tensor is a batch input, but when calculating the loss, it is a constant instead of a vector of batch_size.
The loss returned by the self.optimize function is the average loss of the batch samples (calculated by torch.mean), and the sum of the losses of independent samples. 
![Image](https://github.com/user-attachments/assets/739c4119-9fd1-473b-8f7f-dbb4024b291f)
This leads to the destruction of the independence of each sample when calculating the gradient. I think that when each trajectory is generated during inference, the calculation of the loss of the current environment should be independent, rather than being affected by other trajectories in the same batch.
